src/tracks/Pattern.cpp

```diff
@@ -339,16 +345,27 @@ void Pattern::setType( PatternTypes _new_pattern_type )
 void Pattern::checkType()
 {
 	NoteVector::Iterator it = m_notes.begin();
-	while( it != m_notes.end() )
+
+	// If all notes are StepNotes, we have a BeatPattern
+	bool beatPattern = true;
+	while (it != m_notes.end())
 	{
-		if( ( *it )->length() > 0 )
+		if ((*it)->type() != Note::StepNote)
 		{
-			setType( MelodyPattern );
-			return;
+			beatPattern = false;
+			break;
 		}
 		++it;
 	}
-	setType( BeatPattern );
+
+	if (beatPattern)
+	{
+		setType(BeatPattern);
+	}
+	else
+	{
+		setType(MelodyPattern);
+	}
```

Comment:
`setType(beatPattern ? BeatPattern : MelodyPattern);` is shorter, BTW.

Reply:
Changed to your suggestion. I went for the one line `if` at first because I'm not a fan of using the ternary to run a statement instead of returning a value (as in `beatPattern ? setType(BeatPattern) : setType(MelodyPattern);`), but inside the method call as a way to choose the argument looks cleaner to me and saves a line.